tool/generator/bcd.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert' hide json;
+import 'dart:js_interop';
+
+import 'package:path/path.dart' as p;
+
+import 'filesystem_api.dart';
+
+/// A class to read from the browser-compat-data files and parse interface and
+/// property status (standards track, experimental, deprecated) and supported
+/// browser (chrome, safari, firefox) info.
+class BrowserCompatData {
+  static BrowserCompatData read() {
+    final path =
+        p.join('node_modules', '@mdn', 'browser-compat-data', 'data.json');
+    final content = (fs.readFileSync(
+      path.toJS,
+      JSReadFileOptions(encoding: 'utf8'.toJS),
+    ) as JSString)
+        .toDart;
+
+    final api = (jsonDecode(content) as Map)['api'] as Map<String, dynamic>;
+    final interfaces = api.keys
+        .where((key) => !key.startsWith('_'))
+        .map((key) => BCDInterfaceStatus(key, api[key] as Map<String, dynamic>))
+        .toList();
+    return BrowserCompatData(Map.fromIterable(
+      interfaces,
+      key: (i) => (i as BCDInterfaceStatus).name,
+    ));
+  }
+
+  final Map<String, BCDInterfaceStatus> interfaces;
+
+  BrowserCompatData(this.interfaces);
+
+  BCDInterfaceStatus? interfaceFor(String name) => interfaces[name];
```

Comment:
nit: I like the brevity of the map-y syntax, thoughts on making this a `[]` operator instead here and for other `_For` methods?

Reply:
Fair enough, the name doesn't matter too much for me in that case (honestly I think the `retrieve` is redundant, but either way works).